### PR TITLE
[WIP] Expose file size from TextStorage

### PR DESCRIPTION
**Pull request by amcasey**
The `TextStorage` type is surprisingly stateful, so I'm not sure I've hooked things up correctly.  The goal was to expose the size of the underlying file, either by checking the length of the stored text or by returning the size on disk, if the text is not stored because the file is too large.

I've prepared two different implementations.  I believe `getFileSize` will always return an accurate value, but may touch the disk if a reload is pending.  `getTelemetryFileSize`, on the other hand, will return the last known file size and will only touch the disk if the contents have never been loaded.

Thoughts?

Note: none of this is actually hooked up to anything, but the plan is to expose it through the file info telemetry.

### CODE REVIEW

**Comment by sheetalkamat** [comment]
Actually when I look at this I realise that this is not what you want.  Since your script info could have changed and could be not in sync with the text in the program or for that matter when text is moved to svc, its actual text is never calculated unless must it just works on changes and change range.

I think the accurate information would be from program (except you figure out if this was largeFile by checking if scriptInfo for sourceFile.largeFileSize )

File: src/server/scriptInfo.ts [hunk 1]
```diff
@@ -119,6 +134,23 @@ namespace ts.server {
...
 
+        public getFileSize(): number {
+            return this.useScriptVersionCacheIfValidOrOpen()
+                ? this.svc!.getSnapshot().getLength()
+                : (this.fileSize || this.getOrLoadText().length);
+        }
+
+        public getTelemetryFileSize(): number {
+            // This is only for telemetry, so a slightly stale value is fine.
+            return !!this.fileSize
```
Comment (sheetalkamat) [inline src/server/scriptInfo.ts]: I think this is better and efficient version.
```ts
return this.text !== undefined ? thist.fileSize || this.text.length : Debug.assertDefined(this.svc).getSnapshot().getLength();
```

Comment (amcasey) [reply src/server/scriptInfo.ts]: It looks like that `assertDefined` fails if the size is requested immediately after construction of the `TextStorage`.  I'll check `text` before `svc` though.

**Comment by amcasey** [comment]
Real PR coming soon.

**Pull request closed without merging**